### PR TITLE
[DOCS]/GDOC-108/GDOC-143/Add in Contributing fields and updates

### PR DESCRIPTION
**Pull request by petermoyer**
Solves GDOC-140, GDOC-129, GDOC-147, GDOC-143

Changes proposed in this pull request:
- GDOC-140 Add Glossary
- GDOC-129 Add Contributing sections
- GDOC-147 Remove older style func calls
- GDOC-143 Update links in customize deployment

### Definition of Done
Please delete options that are not relevant.

- [ x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [ x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have made corresponding changes to the documentation
- [x ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: docs/contributing/contributing.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: Contributing
+---
+
+Welcome to the Great Expectations project! We’re very glad you want to help out by contributing.
+
+Our goal is to make your experience as great as possible. Please follow these steps to contribute:
+
+1 Join the community on Slack
```
Comment (Aylr) [inline docs/contributing/contributing.md]:
```suggestion
## 1. Join the community on Slack
```

File: docs/contributing/contributing.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Go to greatexpectations.io/slack and introduce yourself in the #contributors-contributing channel.
+
+2 Set up your development environment
```
Comment (Aylr) [inline docs/contributing/contributing.md]:
```suggestion
## 2. Set up your development environment
```

File: docs/contributing/contributing.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Follow these instructions to set up your dev environment.
+
+Alternatively, for small changes that don’t need to be tested locally (e.g. documentation changes), you can make changes directly through Github.
+
+3 Identify the type of contribution that you want to make
```
Comment (Aylr) [inline docs/contributing/contributing.md]:
```suggestion
## 3. Identify the type of contribution that you want to make
```

File: docs/contributing/contributing.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Issues in GitHub are a great place to start. Check out the help wanted and good first issue labels. Comment to let everyone know you’re working on it.
+
+If there’s no issue for what you want to work on, please create one. Add a comment to let everyone know that you’re working on it. We prefer small, incremental commits, because it makes the thought process behind changes easier to review.
+
+Our Levels of maturity grid provides guidelines for how the maintainers of Great Expectations evaluate levels of maturity of a feature.
+
+4 Start developing
```
Comment (Aylr) [inline docs/contributing/contributing.md]:
```suggestion
## 4. Start developing
```

File: docs/contributing/contributing.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+Make sure to reference the Style Guide and instructions on Testing when developing your code. When your changes are ready, run through our Contribution checklist for pull requests.
+
+Note that if it’s your first contribution, you will need to sign a there is a standard [Contributor license agreement](https://docs.greatexpectations.io/en/0.13.8/contributing/miscellaneous.html#contributing-cla)  (CLA).
```
Comment (Aylr) [inline docs/contributing/contributing.md:27]: This link might not be right since it's pinned to a specific version. Will you double check this?

File: docs/reference/expectations/conditional-expectations.md [hunk 6]
```diff
@@ -113,12 +113,9 @@ If *'row_condition_string'* is a complex expression, it will be split into sever
...
-* :
-  func:`expect_table_columns_to_match_ordered_list <great_expectations.dataset.dataset.Dataset.expect_table_columns_to_match_ordered_list>`
-* :
-  func:`expect_table_column_count_to_be_between <great_expectations.dataset.dataset.Dataset.expect_table_column_count_to_be_between>`
-* :
-  func:`expect_table_column_count_to_equal <great_expectations.dataset.dataset.Dataset.expect_table_column_count_to_equal>`
+* ```expect_column_to_exist```
+* ```expect_table_columns_to_match_ordered_list```
+* ```expect_table_column_count_to_be_between```
+* ```expect_table_column_count_to_equal```
```
Comment (Aylr) [inline docs/reference/expectations/conditional-expectations.md:119]: Thank you for removing the rst syntax. In rst these functioned as links to API docs for the specific expectations. Eventually we will want to bring back this functionality. This is true for all instances of `:func:` removal across other files in this PR.

File: docs/reference/glossary-of-expectations.md [hunk 7]
```diff
@@ -1,5 +1,143 @@
...
+Glossary of Expectations
+This is a list of all built-in Expectations. Expectations are extendable so you can create custom expectations for your data domain! To do so see this article: How to create custom Expectations.
+
+## Dataset
+Dataset objects model tabular data and include expectations with row and column semantics. Many Dataset expectations are implemented using column_map_expectation and column_aggregate_expectation decorators.
+
+Not all expectations are currently implemented for each backend. Please see Table of Expectation Implementations By Backend.
+
+### Table shape
+```expect_column_to_exist```
```
Comment (Aylr) [inline docs/reference/glossary-of-expectations.md:13]: A similar comment here: This cheatsheet page is a Very Important Page for our users and the RTD version links to specific API docs for each of the expectations listed. I feel strongly that this should be a high priority, but not part of this PR as the API docs are not in place yet.

File: docs/reference/glossary-of-expectations.md [hunk 8]
```diff
@@ -1,5 +1,143 @@
----
-title: Glossary of Expectations
----
 
-This is a stub
+Glossary of Expectations
```
Comment (Aylr) [inline docs/reference/glossary-of-expectations.md]: remove this redundant title please

File: docs/contributing/contributing-checklist.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
+Contribution checklist
+Following these instructions helps us make sure the code review and merge process go smoothly.
```
Comment (Shinnnyshinshin) [inline docs/contributing/contributing-checklist.md]: Could you have one more run-through of the formatting and links in this document? I'm seeing a number of broken links and formatting that does not match our RTD site

File: docs/contributing/contributing-github.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
+Making changes directly through Github
+If you want to change documentation, but not code, we suggest using the GitHub markdown editor, which means you don’t have to fork the repo at all. Here’s how you do this:
```
Comment (Shinnnyshinshin) [inline docs/contributing/contributing-github.md]: Formatting and links in this file too

File: docs/contributing/contributing-maturity.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+
+
+
+Features and code within Great Expectations are separated into three levels of maturity: Experimental, Beta, and Production.
+
+   Experimental: Try, but do not rely
+   Beta: Ready for early adopters
+   Production: Ready for general use
+Being explicit about these levels allows us to enable experimentation, without creating unnecessary thrash when features and APIs evolve. It also helps streamline development, by giving contributors a clear, incremental path to create and improve the Great Expectations code base.
```
Comment (Shinnnyshinshin) [inline docs/contributing/contributing-maturity.md:10]: Could you have a look at the formatting of this section and the table below?

File: docs/contributing/contributing-setup.md [hunk 12]
```diff
@@ -0,0 +1,162 @@
+--
+title: Contributing - Setting up your Dev Environment
+---
+
```
Comment (Shinnnyshinshin) [inline docs/contributing/contributing-setup.md]:
```suggestion
---
title: Contributing - Setting up your Dev Environment
---

```

File: docs/contributing/contributing-setup.md [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+GitHub will take you to your forked version of the repository.
+
+2. Clone your fork
+
+Click the green Clone button and choose the SSH or HTTPS URL depending on your setup.
+
+````console
+Copy the URL and run git clone <url> in your local terminal.
+````
```
Comment (Shinnnyshinshin) [inline docs/contributing/contributing-setup.md:30]: could you have a look at the formatting here too?

**Review by Shinnnyshinshin — DISMISSED** [review]
Hi Peter, thanks for putting this PR together. I'm seeing a number of formatting changes that didn't transfer over from RTD (and some broken links that should exist).  Could you have a run-through of what's built in the `https://deploy-preview-2972--knoxpod.netlify.app/docs/contributing/contributing` site and let me know after you've made another round of changes?

**Review by Aylr — CHANGES_REQUESTED** [review]

File: docs/contributing/contributing-checklist.md [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+
+Before submitting a pull request
+Once your code is ready, please go through the following checklist before submitting a pull request.
+
+1. Have you signed the CLA?
```
Comment (Aylr) [inline docs/contributing/contributing-checklist.md]: All the numbered steps should be markdown h2 like `## 1. Have you signed the CLA?`

File: sidebars.js [hunk 15]
```diff
@@ -352,7 +352,14 @@ module.exports = {
...
       items: [
-	    { type: 'doc', id: 'contributing/contributing' }
+	    { type: 'doc', id: 'contributing/contributing' },
+	    { type: 'doc', id: 'contributing/contributing-setup' },
+	    { type: 'doc', id: 'contributing/contributing-checklist' },
+	    { type: 'doc', id: 'contributing/contributing-github' },
+	    { type: 'doc', id: 'contributing/contributing-test' },
+	    { type: 'doc', id: 'contributing/contributing-maturity' },
+	    { type: 'doc', id: 'contributing/contributing-style' },
+	    { type: 'doc', id: 'contributing/contributing-misc' }
```
Comment (Aylr) [inline sidebars.js]: I'm not sure why but these non-human IDs are showing up in the rendered TOC instead of the proper titles. Can you take a look here?

**Review by Aylr — CHANGES_REQUESTED** [review]

File: docs/contributing/contributing-checklist.md (hunk 9 above)
Comment (petermoyer) [reply docs/contributing/contributing-checklist.md]: Moved in later work to this brnahc

File: sidebars.js (hunk 15 above)
Comment (petermoyer) [reply sidebars.js]: The titles were missing

File: docs/contributing/contributing.md (hunk 5 above)
Comment (petermoyer) [reply docs/contributing/contributing.md:27]: Is ok.. line wrapping is funny
Note that if it’s your first contribution, you will need to sign a there is a standard Contributor license agreement (CLA).

File: docs/contributing/contributing-setup.md (hunk 13 above)
Comment (petermoyer) [reply docs/contributing/contributing-setup.md:30]: Looks ok in browser

File: docs/reference/expectations/conditional-expectations.md (hunk 6 above)
Comment (petermoyer) [reply docs/reference/expectations/conditional-expectations.md:119]: Handle in gdoc-156

File: docs/reference/glossary-of-expectations.md (hunk 7 above)
Comment (petermoyer) [reply docs/reference/glossary-of-expectations.md:13]: Handle in gdoc-156

File: docs/contributing/contributing-github.md [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+Please complete the form and comment on the PR to say that you’ve signed the form.
+
+## 5. Verify continuous integration checks
+
+Wait for the other Continuous Integration (CI) checks to go green and watch out for a comment from the automated linter that checks for syntax and formatting issues.
+
+Fix any issues that are flagged. (For documentation changes, it’s unlikely that you’ll have any issues.)
+
+6. Wait for a core team member to approve and merge your PR
```
Comment (Aylr) [inline docs/contributing/contributing-github.md]: Missed a markdown header here

File: docs/contributing/contributing-github.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+Once all checks pass, a Great Expectations team member will approve your PR and merge it.
+
+GitHub will notify you of comments or a successful merge according to your notification settings.
+
+If there are any issues, please address them promptly.
+
+7. Do a victory dance
```
Comment (Aylr) [inline docs/contributing/contributing-github.md]: and here

File: docs/contributing/contributing-misc.md [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+---
+
+Contributor license agreement (CLA)
+When you contribute code, you affirm that the contribution is your original work and that you license the work to the project under the project’s open source license. Whether or not you state this explicitly, by submitting any copyrighted material via pull request, email, or other means you agree to license the material under the project’s open source license and warrant that you have the legal authority to do so.
+
+Please make sure you have signed our Contributor License Agreement (either Individual Contributor License Agreement v1.0 or Software Grant and Corporate Contributor License Agreement (“Agreement”) v1.0).
+
+We are not asking you to assign copyright to us, but to give us the right to distribute your code without restriction. We ask this of all contributors in order to assure our users of the origin and continuing existence of the code. You only need to sign the CLA once.
+
+Release checklist
```
Comment (Aylr) [inline docs/contributing/contributing-misc.md]: Mind making this a heading?

File: docs/contributing/contributing-misc.md [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+
+Review all items on the conda-forge CI system and PR checklist. The active conda-forge community frequently updates build processes and testing, and may help discover issues not observed in our pypi deployment process.
+
+If you need to re-run a failing build and don’t have appropriate permissions or you don’t have permissions to merge please refer to the Account Permissions Overview on the superconductive internal wikiwiki for who to ask. Other conda-forge community partners are extremely responsive and may be able to help resolve issues quickly.
+
+Check for open issues in the GE conda-forge repository. If there are open issues that do not have a corresponding issue in the main GE repo, please create an issue in the GE repo with a link to the corresponding conda issue (e.g. issue #2021). This allows us to internally triage and track the issue.
+
+Celebrate! You have successfully released a new version of Great Expectations!!
+
+Beta Release Notes
```
Comment (Aylr) [inline docs/contributing/contributing-misc.md]: Please make this a heading too

**Review by Aylr — CHANGES_REQUESTED** [review]
A few tiny change requests

**Review by jcampbell — APPROVED** [review]
Simply flagging that all conversations are marked as resolved!
